Add Run Issues to the Test Analytics docs (#2482)
buildkite/docs: pages/test_analytics/test_suites.md

Test Analytics reviews the test results to detect flaky tests after every test run.

## Run issues

<%= image "run-issues.png", alt: "Screenshot of a run with issues displaying in a list, including flaky, slow and failures." %>

> Test Analytics will automatically detect issues per run. For each test, we currently detect up to three issues: flaky, slow and failure, listed in the following order of urgency:

Reviewer: Technically its "number of issues, flaky, slow, failed", I wonder if we should clarify that?

Reviewer 2: @KatieWright26 do you mean that it should be?

>  For each test, we detect flaky, slow and failed issues, along with the total number of issues found. These issues are listed in the following order of urgency:

Author: The confusion here is around the ordering, we say it is ordered 'flaky, slow and failure' where it is actually ordered 'total number of issues, flaky, slow and failure`

Something like:
For each test, we currently detect up to three issues: flaky, slow and failure. These issues are organised and presented in the following order based on their level of urgency, tests with the most amount of issues will be positioned at the top of the list.

Reviewer 2: Still confused, almost there!

The system identifies these 3 issues for tests (the entire test file), and orders the list of test files with the one that has the most issues at the top, decending? eg.
**Worst spec**
`user_authentication_spec.rb`
* Total issues: 12
   * Slow: 4
   * Flaky: 4
   * Failed: 4

**Second worst spec**
`hello_user_spec.rb`
 * Total issues: 6
   * Slow: 2
   * Flaky: 2
   * Failed: 2

Author: It is per execution within test file, so not the whole file. The issues with either have a count of `1 | 0` so like your example:

**Worst spec**
`user_authentication_spec.rb:21`
Total issues: 3
Slow: 1
Flaky: 1
Failed: 1

**Second worst spec**
`user_authentication_spec.rb:2`
Total issues: 2
Slow: 1
Flaky: 0
Failed: 1

Do you think we should include that example in the doc?

> Test Analytics will automatically detect issues per run. For each test, we currently detect up to three issues: flaky, slow and failure, listed in the following order of urgency:

Reviewer 2: ```suggestion
Test Analytics will automatically detect issues for each run. For each test, we currently detect up to three issues: flaky, slow and failure, listed in the following order of urgency:
```

> Test Analytics will automatically detect issues per run. For each test, we currently detect up to three issues: flaky, slow and failure, listed in the following order of urgency:

Reviewer 2: ```suggestion
Test Analytics will automatically detect issues per run. For each test, we currently detect three issues: flaky, slow and failure. And display total number of issues, along with the following, ordered by the most problematic to least problematic:
```

- **Flaky:** [See section on detecting flaky tests](#detecting-flaky-tests).

> - **Slow:** Slowness is measured on the comparative performance of tests within the current run. A slow test is seen as an outlier when it is part of the slowest 1% of tests that took more than 15% of the total run time.

Reviewer 2: This is haaard. Does this make sense still?

```suggestion
- **Slow:** Slowness is measured by the comparative performance of tests within the current run. A test is considered slow when it is in the slowest 1% of tests that took more than 15% of the total run time.
```

Reviewer: It is right?? ChatGPT offered:
> When the slowest 1% of tests extend beyond 15% of the overall run time, our system takes charge. We automatically flag these slower tests, ensuring your testing process stays agile and efficient. 
That feels closer to what we want?

Reviewer 2: I think our human words are far better than ChatGPT's attempt.

How about?
> Slowness is measured by the comparative performance of tests within the current run. The system automatically flags slow tests when the slowest 1% of tests take more than 15% of the overall run time.

> - **Slow:** Slowness is measured on the comparative performance of tests within the current run. A slow test is seen as an outlier when it is part of the slowest 1% of tests that took more than 15% of the total run time.

Reviewer 2: ```suggestion
- **Slow:** Slowness is measured by the comparative performance of tests within the current run. The system automatically flags slow tests when the slowest 1% of tests take more than 15% of the overall run time.
```

Test Analytics reviews the test results to detect flaky tests after every test run.

## Run issues

>🛠 Currently in private beta
> This feature is currently in private beta. If you would like to request access to Run issues, [contact us here](support@buildite.com)

<%= image "run-issues.png", alt: "Screenshot of a run with issues displaying in a list, including flaky, slow and failures." %>

Test Analytics will automatically detect issues for each run. For each test, we currently detect up to three issues: flaky, slow and failure, listed in the following order of urgency:

- **Flaky:** [See section on detecting flaky tests](#detecting-flaky-tests).

- **Slow:** Slowness is measured by the comparative performance of tests within the current run. The system automatically flags slow tests when the slowest 1% of tests take more than 15% of the overall run time. This overall percentage can be manually amended within the suite settings.

> 

Author: @MelissaKaulfuss @KatieWright26 new sentence has been added to the end of this to inform users that the overall percentage can be amended within settings

Reviewer: LGTM!
